File: VirtualLab_MatLab/VirtualLab_init.m


function VirtualLab_init(machine,restore_paths)

    if nargin < 1
        machine = "Tokalab";
        restore_paths = 0;
    elseif nargin < 2
        restore_paths = 0;
    end

    if restore_paths == 1
        disp("restoring default paths")
        restoredefaultpath
    end

    % Directory for VirtualLab
    path_main = fileparts(mfilename('fullpath'));

    paths_to_add = ["\examples";...
        "\SimPla_MATLAB";...
        "\SimPla_MATLAB\functions";...
        "\SimPla_MATLAB\tokamaks";...
        "\SimPla_MATLAB\tokamaks\equilibrium";...
        "\SimPla_MATLAB\tokamaks\geometry";...
        "\SimPla_MATLAB\tokamaks\kinetic";...
        "\SynDiag_MATLAB";...
        "\SynDiag_MATLAB\diagnostics";...
        "\SynDiag_MATLAB\diagnostics\"+machine;...
        "\SynDiag_MATLAB\diagnostics\"+machine+"\diagnostics_data";];

    for i = 1 : length(paths_to_add)
        path_new = path_main + paths_to_add(i);
        if ~contains(path, path_new)
            addpath(path_new);
            fprintf('new added path : %s\n', path_new);
        end
    end

end

File: VirtualLab_Python/VirtualLab_init.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jun  5 10:40:42 2025

@author: ricca
"""

import os
import sys

machine = "Tokalab"

# Ottiene la directory di questo script
path_main = os.path.dirname(os.path.abspath(__file__))

# Percorsi da aggiungere
paths_to_add = [
     "examples",
        "SimPla_Python",
        "SimPla_Python/functions",
        "SimPla_Python/tokamaks",
        "SimPla_Python/tokamaks/equilibrium",
        "SimPla_Python/tokamaks/geometry",
        "SimPla_Python/tokamaks/kinetic",
        "SynDiag_Python",
        "SynDiag_Python/diagnostics",
        f"SynDiag_Python/diagnostics/{machine}",
        f"SynDiag_Python/diagnostics/{machine}/diagnostics_data"
    ]

# Aggiunge i percorsi a sys.path se non già presenti
for relative_path in paths_to_add:
    full_path = os.path.join(path_main, relative_path)
    if full_path not in sys.path:
        sys.path.append(full_path)
        print(f"new added path : {full_path}")